app/src/main/java/org/dhis2/uicomponents/map/managers/EventMapManager.kt

```diff
@@ -65,6 +107,31 @@ class EventMapManager(mapView: MapView) : MapManager(mapView) {
         }
     }
 
+    override fun findFeatures(
+        source: String,
+        propertyName: String,
+        propertyValue: String
+    ): List<Feature>? {
+        return mutableListOf<Feature>().apply {
+            featureCollection?.features()?.filter {
+                it.getStringProperty(propertyName) == propertyValue
+            }?.map {
+                mapLayerManager.getLayer(LayerType.EVENT_LAYER.name)
+                    ?.setSelectedItem(it)
+                it
+            }?.let { addAll(it) }
+            deFeatureCollection.values.map { collection ->
+                collection.features()?.filter {
+                    mapLayerManager.getLayer(it.getStringProperty(DE_NAME))?.visible == true &&
+                        it.getStringProperty(propertyName) == propertyValue
+                }?.map {
+                    mapLayerManager.getLayer(it.getStringProperty(DE_NAME))?.setSelectedItem(it)
+                    it
+                }?.let { addAll(it) }
+            }
+        }
+    }
```

Comment:
Do you think we could test it? (Just asking hehe)

Reply:
Answer in next comment!